### PR TITLE
feat(MCDA): 19032 Append units to custom axis labels in MCDA dropdown

### PR DESCRIPTION
**Pull request by albaranau**
https://kontur.fibery.io/Tasks/Task/FE-MCDA-dropdown-add-units-to-custom-axis-labels-19032

## Summary by CodeRabbit

- **New Features**
	- Introduced a custom label formatting function for bivariate axes, allowing more descriptive and flexible labeling.
	- Updated logic for axis label generation to better handle custom labels and associated quotients.

- **Bug Fixes**
	- Improved the label generation process to ensure accurate formatting when custom labels are used.

- **Tests**
	- Added comprehensive unit tests for the new custom label formatting function, validating various scenarios for accurate label representation.

### CODE REVIEW

File: src/utils/bivariate/labelFormatters.ts [hunk 1]
```diff
@@ -10,6 +10,17 @@ export const convertDirectionsArrayToLabel = (directions: string[][]) => {
   return `${formatSentimentDirection(from)} → ${formatSentimentDirection(to)}`;
 };
 
+export const formatCustomLabelForBivariateAxis = (
+  customLabel: string,
+  quotients: Axis['quotients'],
+): string => {
+  const units = formatBivariateAxisUnit(quotients);
```
Comment (Natallia-Harshunova) [inline src/utils/bivariate/labelFormatters.ts:17]: Just can be simplified
```
const units = formatBivariateAxisUnit(quotients);
return units ? `${customLabel} (${units})` : customLabel;
```

File: src/utils/bivariate/helpers/converters/axisDTOtoAxis.ts [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
 import type { AxisDTO } from '~core/resources/bivariateStatisticsResource/types';
 import type { Axis } from '~utils/bivariate/types/stat.types';
 
 export function axisDTOtoAxis(dto: AxisDTO): Axis {
   return {
     ...dto,
     id: dto.quotient.join('|'),
-    label: dto.label || formatBivariateAxisLabel(dto.quotients),
+    label: dto.label
+      ? formatCustomLabelForBivariateAxis(dto.label, dto.quotients)
```
Comment (Natallia-Harshunova) [inline src/utils/bivariate/helpers/converters/axisDTOtoAxis.ts]: maybe for consistency name function as `formatBivariateAxisCustomLabel` to make difference more clear?

**Review by Natallia-Harshunova — APPROVED** [review]

File: src/utils/bivariate/helpers/converters/axisDTOtoAxis.ts (hunk 2 above)
Comment (albaranau) [reply src/utils/bivariate/helpers/converters/axisDTOtoAxis.ts]: fixed

File: src/utils/bivariate/labelFormatters.ts (hunk 1 above)
Comment (albaranau) [reply src/utils/bivariate/labelFormatters.ts:17]: fixed

**Pull request merged**